ahcore/writers.py

```diff
@@ -205,6 +212,12 @@ def consume(
         try:
             with h5py.File(self._filename.with_suffix(".h5.partial"), "w") as h5file:
                 first_coordinates, first_batch = next(batch_generator)
+
+                if self._precision:
```

Comment:
Since you have the same code below, you could decide to place this into a small method like
```
def adjust_batch_precision(batch: Tensor) -> Tensor:
    "your code here"
    return batch
```
Then in both cases you can call `batch = self.adjust_batch_precision(batch)`.

Reply:
Ah that's a good one. How silly of me!